cmd/cluster-agent/api/v1/clusterchecks.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build clusterchecks
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks"
+)
+
+var clusterCheckHandler *clusterchecks.Handler
+
+// Install registers v1 API endpoints
+func installClusterCheckEndpoints(r *mux.Router) {
+	r.HandleFunc("/clusterchecks/allconfigs", getAllCheckConfigs).Methods("GET")
+	// TODO
+	//r.HandleFunc("/clusterchecks/status/{nodeName}", postCheckStatus).Methods("POST")
+	//r.HandleFunc("/clusterchecks/configs/{nodeName}", getCheckConfigs).Methods("GET")
+}
+
+func SetClusterCheckHandler(h *clusterchecks.Handler) {
+	clusterCheckHandler = h
+}
```

Comment:
@devonboyer I like that. Cannot find `ServerContext` as a concrete type in our codebase, could you please elaborate on how you'd do that?

Reply:
@xvello 

That was probably a bit unclear, the `ServerContext` doesn't exist yet :P

I think it would make sense to create a new type (possibly called `ServerContext`) to store any information that might be needed in install the endpoints. This could be instantiated in `StartServer` and passed to `SetupHandlers`. For your case this type would hold a pointer to `clusterchecks.Handler`.

The `ServerContext` solves the problem of having global variables to hold all the "manager" instances in the agent.